infrastructure/src/main/java/org/corfudb/infrastructure/log/StreamLogFiles.java

```diff
@@ -1012,6 +1018,35 @@ private boolean verify(List<LogData> range) {
         return endSegment - firstSegment <= 1;
     }
 
+    /**
+     * This method requests for missing addresses in this Log Unit in the specified consecutive
+     * range of addresses. The range cannot be larger than RECORDS_PER_LOG_FILE.
+     *
+     * @param range Range of addresses to compare.
+     * @return Set of missing addresses.
+     */
+    @Override
+    public Set<Long> getMissingAddresses(Range<Long> range) {
```

Comment:
please simplify

```java

    public List<Long> rangeQuery(Range<Long> range) {
        List<Long> ret = new ArrayList<>(1000);
        for (long a = range.lowerEndpoint(); a <= range.upperEndpoint(); a++) {
            if (getSegmentHandleForAddress(a).getKnownAddresses().containsKey(a)) {
                ret.add(a);
            }
        }
        return ret;
    }
```

Reply:
This would acquire a lock, every time you call `getSegmentHandleForAddress`.
I can work on simplifying this.

Reply:
Done.